base/threadingconstructs.jl

```diff
@@ -22,8 +22,22 @@ See also: `BLAS.get_num_threads` and `BLAS.set_num_threads` in the
 """
 nthreads() = Int(unsafe_load(cglobal(:jl_n_threads, Cint)))
 
-function threading_run(func)
+function prefer_threaded_region(f)
```

Comment:
Maybe a better name is

```suggestion
function prefer_io(f)
```

?

I also wonder if it makes sense to expose this as a public API with the caveat that it is just a hint and `f -> f()` is a valid implementation of this. It then maybe even makes sense to not do this in `@threads for` by default to help CPU-bound (almost no I/O) use cases since people can recover the old behavior with `prefer_io() do; @threads for ... end; end`.